packages/engine/src/scene/components/RenderSettingsComponent.ts

```diff
@@ -93,6 +95,24 @@ export const RenderSettingsComponent = defineComponent({
       renderer.renderer.shadowMap.needsUpdate = true
     }, [component.shadowMapType])
 
+    useEffect(() => {
+      const directionalLightEntities = directionalLights
+      let primaryLight = UndefinedEntity
+      for (const directionalLightEntity of directionalLightEntities) {
+        if (component.primaryLight.value === getComponent(directionalLightEntity, UUIDComponent)) {
+          primaryLight = directionalLightEntity
+          break
+        }
+      }
+
+      if (!primaryLight && component.primaryLight.value) {
+        component.merge({
```

Comment:
do we actually want to mutate this here? If the data is invalid it should just do nothing